pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -268,20 +268,28 @@ public String recommendConfig(String inputStr) {
   @Path("/tables")
   @Authorize(targetType = TargetType.CLUSTER, action = Actions.Cluster.GET_TABLE)
   @ApiOperation(value = "Lists all tables in cluster", notes = "Lists all tables in cluster")
-  public String listTables(@ApiParam(value = "realtime|offline") @QueryParam("type") String tableTypeStr,
+  public String listTables(@ApiParam(value = "realtime|offline|dimension") @QueryParam("type") String tableTypeStr,
       @ApiParam(value = "Task type") @QueryParam("taskType") String taskType,
       @ApiParam(value = "name|creationTime|lastModifiedTime") @QueryParam("sortType") String sortTypeStr,
       @ApiParam(value = "true|false") @QueryParam("sortAsc") @DefaultValue("true") boolean sortAsc) {
     try {
+      final boolean isDimensionTable = "dimension".equalsIgnoreCase(tableTypeStr);
       TableType tableType = null;
-      if (tableTypeStr != null) {
+      if (isDimensionTable) {
+        // Dimension is a property (isDimTable) of an OFFLINE table.
+        tableType = TableType.OFFLINE;
+      } else if (tableTypeStr != null) {
         tableType = TableType.valueOf(tableTypeStr.toUpperCase());
       }
       SortType sortType = sortTypeStr != null ? SortType.valueOf(sortTypeStr.toUpperCase()) : SortType.NAME;
 
-      List<String> tableNamesWithType = tableType == null ? _pinotHelixResourceManager.getAllTables()
-          : (tableType == TableType.REALTIME ? _pinotHelixResourceManager.getAllRealtimeTables()
-              : _pinotHelixResourceManager.getAllOfflineTables());
+      // If tableTypeStr is dimension, then tableType is set to TableType.OFFLINE.
+      // So, checking the isDimensionTable to get the list of dimension tables only.
+      List<String> tableNamesWithType =
+          isDimensionTable ? _pinotHelixResourceManager.getTableCache().getAllDimensionTables()
```

Comment:
You may still add `getAllDimensionTables()` into `PinotHelixResourceManager` which can be a useful helper method

Reply:
:+1: we can filter out those instead of pulling all out of zk then filter ourselves

Reply:
> 👍 we can filter out those instead of pulling all out of zk then filter ourselves

I am using `TableCache` to avoid pulling from ZK, is there still some for improvement?